library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -47,10 +47,19 @@
  * @since 3.2.0
  */
 @InterfaceStability.Evolving
-public abstract class Parser<INPUT> extends BaseOperator implements Converter<INPUT, Object>,
-    ActivationListener<Context>
+public abstract class Parser<INPUT> extends BaseOperator implements Converter<INPUT, Object>
 {
   protected transient Class<?> clazz;
+  @AutoMetric
+  protected long errorTupleCount;
+  @AutoMetric
+  protected long emittedObjectCount;
+  @AutoMetric
+  protected long rollingPercentValidTuples;
+  @AutoMetric
+  protected long rollingPercentErrorTuples;
+  @AutoMetric
+  protected long incomingTuplesCount;
 
   @OutputPortFieldAnnotation(schemaRequired = true)
```

Comment:
@ishark @chandnisingh Yes, you are correct. I tested it again. No need to remove the annotation